pandas/_libs/algos.pyx

```diff
@@ -983,6 +989,9 @@ def rank_1d(
     else:
         mask = np.zeros(shape=len(masked_vals), dtype=np.uint8)
 
+    # If ascending and na_option == 'bottom' or descending and
+    # na_option == 'top' -> we want to rank NaN as the highest
+    # so fill with the maximum value for the type
```

Comment:
just want to highlight https://github.com/pandas-dev/pandas/pull/40016 in case relevant here for consistency

Reply:
Thanks for linking that, I was confused by the documentation as well haha. Will adjust this comment in #40546